### PR TITLE
Login & Edit UI

### LINKED ISSUE #106: First pass at inline editing content (3)
As Nova, I need to be able to make edits to content, so that I can share my experiences with the community.

**ACCEPTANCE CRITERIA**
**Given** I am viewing a practice
**When** I select to edit that practice
**Then** I can view the edit interface inline with the content

**NOTES**
- This is a first pass at an unvalidated, MVP concept. Also, it's technically not a Nova story.

**BLOCKERS**

### PR DESCRIPTION
**Pull request by shaheinm**
Resolves #106 

For the first go-round, I'm using [Outline's](https://github.com/outline/rich-markdown-editor) inline rich markdown editor. Markdown works more like a shortcut there, and you have a rich-text-like editor interface. If you double click on text, you'll see a formatting pop-up. If you're on a new line, you'll see a plus sign (or you can type `/`), and you'll get a bunch of formatting options.  You can also use typical keyboard shortcuts (like ctrl+b for bold).

I also included another editor called [Draftail](https://www.draftail.org/), which is a rich text editor built on top of Facebook's Draft.js project. It's specifically built to work well with React, but as a rich text editor, I have to convert Markdown to Rich Text to edit, and convert it back to Markdown to save.

I was able to complete that, but the Draft.js editor does not include things like links and images, which the outline editor does (we can eventually add a function to upload images through the media service directly through the editor -- for inline images).

In order to see the Edit button on the Practice Page, you will have to log in. If you've contributed to the Open Practice Library before, you're already in the Strapi users list (ping me for the password if you want to test). If you haven't, you'll have to ask me to or add yourself to the Strapi users list. There is no registration for v1 (new content contributors will need to request a login through our team directly).

The login flow is a good example of how to use React's context API, by the way.

I also fixed the test with the weird date, and cleaned up some stuff in the icon components that were causing tests to log problems (without failing, which is because we're only doing snapshot tests there -- **bad developer**).

### CODE REVIEW

**Comment by shaheinm** [comment]
## How to test

Click the login button (Home/Landing and Practice Page):
image

Enter credentials and hit enter or click the button. If you do it from the Home/Landing page, you'll be redirected to the All Practices/Home page.

If you logged from a Practice Page, the page will refresh with an edit button next to the title. Otherwise, go to a Practice Page and see the edit button:
image

Click the edit button. It will turn into two buttons (cancel and save):
image

You should also notice the title and subtitle are underlined and editable now. In the page body, the content areas have a grey background to indicate you're in edit mode:
image

The image above also shows the formatting toolbar that pops up if you highlight any text.  Here is the new line box I mentioned (click the + sign or type `/`):
image

You can also edit links by clicking on them:
image

One last thing that I thought I was exciting -- Outline's editor has integrations available, and one of them is Miro -- check them all [here](https://www.getoutline.com/integrations).

File: src/components/practicePage/PageIntro/__tests__/PageIntro.test.js [hunk 1]
```diff
@@ -7,7 +7,8 @@ import PageIntro from '../index';
 
 afterEach(cleanup);
 
-const today = Date.now();
+const date = new Date(2020,6,8);
```
Comment (lallmon) [inline src/components/practicePage/PageIntro/__tests__/PageIntro.test.js:10]: external-content duckduckgo com

LOL

File: src/components/shared/Login/LoginButton.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  },
+  drawerPaper: {
+    display: "flex",
+    width: drawerWidth,
+    backgroundColor: theme.palette.common.true_white,
+    alignContent: "center",
+    alignItems: "center",
+  },
+  loginButton: {
+    height: "34px",
```
Comment (lallmon) [inline src/components/shared/Login/LoginButton.js:28]: Where do these magic number values come from? Is anyone on the dev team opposed to using 16 pix REM values, for simple math and browser zoom scalability?

Comment (shaheinm) [reply src/components/shared/Login/LoginButton.js:28]: I would prefer to use rem but I got those PX numbers from zeplin

Comment (lallmon) [reply src/components/shared/Login/LoginButton.js:28]: That was my hunch. I've seen REM in the codebase, do you know if Material bases the REM at 16?

**Pull request merged**